WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,18 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    self.rounds.each(&:destroy!)
+    self.reload
+    self.rounds_attributes = wcif["rounds"].map.with_index do |wcif_round, index|
+      round_number = index + 1
+      Round.wcif_to_round_attributes(wcif_round, round_number)
+    end
+    self.save!
+
+    if wcif["rounds"].empty?
```

Comment:
It's not the only way to delete an event, you could also remove the event at the next level up in the WCIF. This is just to keep some consistency: I don't see the value in having events with no rounds.